lib/components/ArtistScreen/album_list_tile.dart

```diff
@@ -0,0 +1,583 @@
+import 'package:audio_service/audio_service.dart';
+import 'package:finamp/models/finamp_models.dart';
+import 'package:finamp/screens/artist_screen.dart';
+import 'package:finamp/services/queue_service.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_gen/gen_l10n/app_localizations.dart';
+import 'package:flutter_tabler_icons/flutter_tabler_icons.dart';
+import 'package:get_it/get_it.dart';
+
+import '../../models/jellyfin_models.dart';
+import '../../services/audio_service_helper.dart';
+import '../../services/jellyfin_api_helper.dart';
+import '../../services/finamp_settings_helper.dart';
+import '../../services/downloads_helper.dart';
+import '../../services/process_artist.dart';
+import '../../services/music_player_background_task.dart';
+import '../../screens/album_screen.dart';
+import '../../screens/add_to_playlist_screen.dart';
+import '../AlbumScreen/downloaded_indicator.dart';
+import '../favourite_button.dart';
+import '../album_image.dart';
+import '../print_duration.dart';
+import '../error_snackbar.dart';
+
+enum AlbumListTileMenuItems {
+  addFavourite,
+  removeFavourite,
+  addToMixList,
+  removeFromMixList,
+  playNext,
+  addToNextUp,
+  shuffleNext,
+  shuffleToNextUp,
+  addToQueue,
+  shuffleToQueue,
+}
+
+class AlbumListTile extends StatefulWidget {
+  const AlbumListTile(
+      {Key? key,
+      required this.item,
+
+      /// Children that are related to this list tile, such as the other songs in
+      /// the album. This is used to give the audio service all the songs for the
+      /// item. If null, only this song will be given to the audio service.
+      this.children,
+
+      /// Index of the song in whatever parent this widget is in. Used to start
+      /// the audio service at a certain index, such as when selecting the middle
+      /// song in an album.
+      this.index,
+      this.parentId,
+      this.parentName,
+      this.onDelete})
+      : super(key: key);
+
+  final BaseItemDto item;
+  final List<BaseItemDto>? children;
+  final int? index;
+  final String? parentId;
+  final String? parentName;
+  final VoidCallback? onDelete;
+
+  @override
+  State<AlbumListTile> createState() => _AlbumListTileState();
+}
+
+class _AlbumListTileState extends State<AlbumListTile> {
+  final _audioServiceHelper = GetIt.instance<AudioServiceHelper>();
+  final _queueService = GetIt.instance<QueueService>();
+  final _audioHandler = GetIt.instance<MusicPlayerBackgroundTask>();
+  final _jellyfinApiHelper = GetIt.instance<JellyfinApiHelper>();
+
+  @override
+  Widget build(BuildContext context) {
+    final screenSize = MediaQuery.of(context).size;
+
+    final listTile = ListTile(
+      leading: AlbumImage(item: widget.item),
+      title: RichText(
+        text: TextSpan(
+          children: [
+            // third condition checks if the item is viewed from its album (instead of e.g. a playlist)
+            // same horrible check as in canGoToAlbum in GestureDetector below
```

Comment:
Seems like some leftover comments. You can also remove everything that is underlined in yellow or blue, as that is not being used at all